### ISSUE TITLE
save_image does not specify the format of the images to be saved

### ISSUE THREAD

**Issue by w4nderlust** [pytorch/vision#5211]
### 📚 The doc issue

By looking at the code, it seems like the `save_image` function expects floats tensors with values between 0 and 1, but in the docs that is nowhere to be found. A very reasonable alternative for the user (which is actually what I was initially providing, and what is expected by `sklearn.io.imsave` for instance) is a tensor of uint8 with values between 0 and 255 (which is actually what internally `save_image` casts to).

Please add this expected tensor format to the docs.
Ideally also adding support for passing directly the uint8 would be great as it would make porting from other libraries much simpler than taking the uint8 tensor, transforming it to float and passing it to torchvision just to be caster again to uint8 internally.

Reference line from `save_image`:
`7933    ndarr = grid.mul(255).add_(0.5).clamp_(0, 255).permute(1, 2, 0).to('cpu', torch.uint8).numpy()`

**Comment by NicolasHug** [comment]
Thanks for the report @w4nderlust , you're right, `save_image` could benefit from more detailed docs.

It's also probably not what most users are looking for, considering that it internally relies on `make_grid()` which will add padding pixels by default (and it has a fairly unconventional API: its `nrow` parameter specifies... a number of columns).

I think we should deprecate both eventually, but that's another story.

For your specific use-case, perhaps [write_jpeg()](https://pytorch.org/vision/stable/io.html#torchvision.io.write_jpeg) or [write_png()](https://pytorch.org/vision/stable/io.html#torchvision.io.write_png) would be good alternatives? These accept uint8 tensors with a `C H W` layout, and they're better documented.

**Comment by w4nderlust** [comment]
@NicolasHug thanks for the pointer, those functions are indeed much closer to what i was looking for!
